src/services/emailService.ts

```diff
@@ -0,0 +1,46 @@
+import { none, Option, some } from "fp-ts/lib/Option";
+import * as nodemailer from "nodemailer";
+
+export interface ITransporterOptions {
+  auth: {
+    pass: string;
+    user: string;
+  };
+  host: string;
+  port: number;
+  secure: boolean;
+  from: string;
+}
+
+interface ITransporter {
+  // tslint:disable-next-line:no-any
+  sendMail: (mailOptions: IMailOptions) => Promise<any>;
```

Comment:
why not using nodemailer typings directly ?

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nodemailer/lib/mailer/index.d.ts#L186

Reply:
I didin't use directly nodemailer typings because they are exported from a submodule, and according to the lint rules the import of any submodule is disallowed. Reading the rationale for [this rule](https://palantir.github.io/tslint/rules/no-submodule-imports/), I thought it could be safer to define our specific typing.

Reply:
may we try referencing them from the top level import like:

```ts
import * as NodeMailer from "nodemailer";

...

... NodeMailer.Transporter ...
... NodeMailer.SendMailOptions ...
... NodeMailer.SentMessageInfo ...

```

?